### PR TITLE
Fixed processing of invalid form submissions

### PR DESCRIPTION
**Pull request by ambrozic**
## Description of change
Invalid data from forms was causing tasks to break and enquiries were left unprocessed since October. Invalid forms are now ignored following business decision.

### CODE REVIEW

File: app/enquiries/tests/test_adobe_integration.py [hunk 1]
```diff
@@ -142,6 +143,39 @@ def test_process_second_qualification(self, mock_staging, mock_wf, mock_as, mock
         )
         client.start_workflow.assert_called_with(settings.ADOBE_STAGING_WORKFLOW)
 
+    @freeze_time()
+    @mock.patch('app.enquiries.common.adobe.AdobeClient.get_token')
+    @mock.patch('app.enquiries.common.as_utils.hawk_request')
+    @mock.patch('app.enquiries.common.email_campaign_utils.get_new_second_qualification_forms')
+    def test_process_second_qualification_invalid_data(self, mock_as, mock_token, mock_ecu):
```
Comment (tomkingston) [inline app/enquiries/tests/test_adobe_integration.py:150]: Are mock_token and mock_ecu the wrong way around here?

**Review by tomkingston — COMMENTED** [review]
Looks good, but I think the arguments passed from your mock patches are in the wrong order - let me know if I've got that wrong

File: app/enquiries/tests/test_adobe_integration.py (hunk 1 above)
Comment (ambrozic) [reply app/enquiries/tests/test_adobe_integration.py]: changed tests a little. should be ok now.

**Review by tomkingston — APPROVED** [review]

File: app/enquiries/tests/test_adobe_integration.py [hunk 2]
```diff
@@ -142,6 +143,48 @@ def test_process_second_qualification(self, mock_staging, mock_wf, mock_as, mock
...
+    @mock.patch('app.enquiries.common.adobe.AdobeClient.get_token')
+    @mock.patch('app.enquiries.common.as_utils.hawk_request')
+    def test_process_second_qualification_invalid_data(self, mock_as, mock_token):
+        for emt_id, phone_number, arrange_callback, was_called in [
+            ('None', '0771231234', 'yes', True),
+            ('-1', '0771231234', 'yes', True),
+            (1.2, '0771231234', 'yes', True),
+            (None, '0771231234', 'yes', False),
+            ("", '0771231234', 'yes', False),
+        ]:
```
Comment (tomkingston) [inline app/enquiries/tests/test_adobe_integration.py:156]: You could use pytest.parametrize here if you wanted: https://docs.pytest.org/en/stable/parametrize.html

Comment (ambrozic) [reply app/enquiries/tests/test_adobe_integration.py:156]: i think parametrize does not work with `django.test.TestCase`

Comment (tomkingston) [reply app/enquiries/tests/test_adobe_integration.py:156]: Also, given that `arrange_callback` is yes every time, and the `phone_number` is also the same in each case, perhaps they could be moved out of the fixture

File: app/enquiries/tests/test_adobe_integration.py [hunk 3]
```diff
@@ -142,6 +143,48 @@ def test_process_second_qualification(self, mock_staging, mock_wf, mock_as, mock
...
+                                            'arrange_callback': arrange_callback,
+                                        }
+                                    }
+                                }
+                            }
+                        ]
+                    }
+                }
+                campaign.process_second_qualifications()
+                assert mock_update.called is was_called
```
Comment (mforner13) [inline app/enquiries/tests/test_adobe_integration.py:180]: Could this be asserted using `assert mock_udpate.called`? It could be confusing using the `was_called` value, as that is referring to a physical phone call rather than a function call, and seems unrelated. Unless that function only ever gets called if the contact was called in real life?

Comment (ambrozic) [reply app/enquiries/tests/test_adobe_integration.py:181]: `was_called` refers to mocked function being called or executed and `mock_udpate.called` has to be asserted as a boolean

Comment (mforner13) [reply app/enquiries/tests/test_adobe_integration.py:181]: Understood - much clearer now using booleans in the fixture data! 👍

**Review by mforner13 — APPROVED** [review]
